### DISCUSSION TITLE
Android Auto

### DISCUSSION THREAD

**Discussion by fcaronte** [maxrave-dev/SimpMusic#207] [Ideas]
Today I discovered this app seems promising and I would to ask if possible to work on the android auto side where actually can show only preferred and nothing other, if try to search something from the car will get no results will be useful if can become an Spotify alternative on the car side too

**Comment by maxrave-dev** [comment]
SimpMusic only show all of track, playlist wwhich saved to phone (local data) database. I am trying to make a online search for Android Auto

**Reply by wazer1** [reply]
Wonder how this person gets SimpMusic to show up in Android Auto instead of just on a phone?
https://www.youtube.com/watch?v=HYw385h-cM0

**Reply by fcaronte** [reply]
As I see it show only your favourite song you cannot Search on all database

**Reply by maxrave-dev** [reply]
> Wonder how this person gets SimpMusic to show up in Android Auto instead of just on a phone?
> https://www.youtube.com/watch?v=HYw385h-cM0

SimpMusic is support Android Auto. Try it now

**Reply by maxrave-dev** [reply]
> As I see it show only your favourite song you cannot Search on all database 

Show all tracks which played in your phone (because played tracks have been saved in databases)

**Reply by m1ghty-mike** [reply]
Which version of Android Auto are you using? My car is giving an error message with the app.

**Reply by Mthomeassistant** [reply]
I'd love it on android auto but it doesn't show in list of apps, is there anything else you have to do? 
Tried the latest and nightly version.

**Reply by Allzweck** [reply]
> I'd love it on android auto but it doesn't show in list of apps, is there anything else you have to do? Tried the latest and nightly version.

A little late, but maybe helpful for future searchers: Enable Developer options in Android Auto(Go to Settings -> Apps & Notifications -> All Apps -> Android Auto -> Advanced -> Additional settings in the app.
Then tap Version and Permission Info 10 times. ) in the 3 dot settings you can then add Unknown Sources, check this and SimpMusic works

**Reply by fcaronte** [reply]
Yes I was able to see the app but I would to know why seem only Spotify app can be a real working app on android auto, all other music app I'm going to try have very limited use through android auto, I don't know of Spotify have special permission or other app are not use all supporterd feature correctly, I talk about navigation in playlist, top 100 and search by category, all other music app have only own playlists or very little other things

**Comment by andry23jsv** [comment]
I implemented and tested a structured Android Auto search experience on a private Metrolist fork that might also be useful for SimpMusic.

Instead of returning a flat list of results, Android Auto shows browsable search categories:

Search: "Gigi D'Agostino"

→ Top results
→ Artists
→ Albums
→ Songs
→ Playlists

Each category is a browsable MediaItem. Opening "Albums", for example, loads the matching albums; opening "Artists" loads artists, and so on.

The flow is basically:

Search query
→ browsable category nodes
→ selected category
→ corresponding online search results

I've tested this successfully with Android Auto / DHU.

I'm not suggesting the Metrolist code can simply be copied, since SimpMusic has its own implementation, but the same MediaLibrary browsing approach should be applicable here.

My fork is private, so I can't link it, but I can share the relevant implementation logic or a reduced diff if it would be useful.
